src/modules/newtek/producer/newtek_ndi_producer.cpp

```diff
@@ -0,0 +1,305 @@
+                    video_frame.frame_format_type == NDIlib_frame_format_type_interleaved ? 1 : 0;
+                av_frame->top_field_first = av_frame->interlaced_frame;
+                NDIlib_audio_frame_interleaved_32s_t audio_frame_32s;
+                audio_frame_32s.p_data = new int32_t[audio_frame.no_samples * audio_frame.no_channels];
+                if (audio_frame.p_data != nullptr) {
+                    audio_frame_32s.reference_level = 0;
+                    ndi_lib_->NDIlib_util_audio_to_interleaved_32s_v2(&audio_frame, &audio_frame_32s);
+                    a_frame->channels    = audio_frame_32s.no_channels;
+                    a_frame->sample_rate = audio_frame_32s.sample_rate;
+                    a_frame->nb_samples  = audio_frame_32s.no_samples;
+                    a_frame->data[0]     = reinterpret_cast<uint8_t*>(audio_frame_32s.p_data);
+                }
+                auto mframe =
+                    ffmpeg::make_frame(this, *(frame_factory_.get()), std::move(av_frame), std::move(a_frame));
+                auto dframe = core::draw_frame(std::move(mframe));
+                {
+                    std::lock_guard<std::mutex> lock(frames_mutex_);
+                    frames_.push(dframe);
+                    while (frames_.size() > 2) { // should never happen because frame sync takes care of it
+                        frames_.pop();
+                        graph_->set_tag(diagnostics::tag_severity::WARNING, "dropped-frame");
+                        CASPAR_LOG(info) << print() << "Frame dropped";
+                    }
+                }
+                ndi_lib_->NDIlib_framesync_free_audio(ndi_framesync_, &audio_frame);
+                ndi_lib_->NDIlib_framesync_free_video(ndi_framesync_, &video_frame);
+                delete[] audio_frame_32s.p_data;
+            }
+
+            graph_->set_value("frame-time", frame_timer_.elapsed() * format_desc_.fps * 0.5);
+        } catch (...) {
+            CASPAR_LOG_CURRENT_EXCEPTION();
+            return false;
+        }
+        return true;
+    }
+
+    // frame_producer
+
+    void initialize()
+    {
+        ndi_lib_     = ndi::load_library();
+        auto sources = ndi::get_current_sources();
+
+        NDIlib_recv_create_v3_t NDI_recv_create_desc;
+        NDI_recv_create_desc.allow_video_fields = false;
+        NDI_recv_create_desc.bandwidth = low_bandwidth_ ? NDIlib_recv_bandwidth_lowest : NDIlib_recv_bandwidth_highest;
+        NDI_recv_create_desc.color_format = NDIlib_recv_color_format_BGRX_BGRA;
+        std::string src_name              = u8(name_);
+
+        auto found_source = sources.find(src_name);
+        if (found_source != sources.end()) {
+            NDI_recv_create_desc.source_to_connect_to = found_source->second;
+        } else {
+            NDI_recv_create_desc.source_to_connect_to.p_ndi_name = src_name.c_str();
+        }
+        NDI_recv_create_desc.p_ndi_recv_name = src_name.c_str();
```

Comment:
That's what documentation says about `p_ndi_recv_name`:
> 	// The name of the NDI receiver to create. This is a NULL terminated UTF8 string and should be 
	// the name of receive channel that you have. This is in many ways symettric with the name of
	// senders, so this might be "Channel 1" on your system. If this is NULL then it will use the 
	// filename of your application indexed with the number of the instance number of this receiver.

Although I can't find any uses of it in any other place in the SDK, I think setting it to "CasparCG `instance_no_`" would be enough.

Reply:
Searching the examples gives some results:
![image](https://user-images.githubusercontent.com/1327476/53831188-24f45e00-3f7c-11e9-93f9-d63594540e4c.png)

I am happy for this to be "CasparCG `instance_no_`"